types/time.go

```diff
@@ -1527,7 +1527,7 @@ func extractSingleTimeValue(unit string, format string) (int64, int64, int64, fl
 	if err != nil {
 		return 0, 0, 0, 0, ErrIncorrectDatetimeValue.GenWithStackByArgs(format)
 	}
-	iv := int64(fv + 0.5)
+	iv := int64(math.Round(fv))
```

Comment:
Should we use `floatStrToIntStr` instead of `Round`? `floatStrToIntStr` does round operation also and it covers corner cases.

Reply:
The float value is also needed.
`iv` is used in cases that float value should be rounded, like `DAY`, `WEEK`.

Reply:
You mean inputs like `1.23a4 DAY` ? `1.23a` can pass `strconv.ParseFloat` @eurekaka  
or something like `1e9223372036854775807` ?

I think the change here is ok, a sane person will not use those illegal input, so 
strconv.ParseFloat + math.Round is suitable for most of the cases

Reply:
```sql
MySQL(root@127.0.0.1:test) > select adddate("2018-12-09", interval "1e1" DAY);
+-------------------------------------------+
| adddate("2018-12-09", interval "1e1" DAY) |
+-------------------------------------------+
| 2018-12-10                                |
+-------------------------------------------+
1 row in set, 1 warning (0.00 sec)

MySQL(root@127.0.0.1:test) > select adddate("2018-12-09", interval "10" DAY);
+------------------------------------------+
| adddate("2018-12-09", interval "10" DAY) |
+------------------------------------------+
| 2018-12-19                               |
+------------------------------------------+
1 row in set (0.00 sec)

MySQL(root@127.0.0.1:test) > select adddate("2018-12-09", interval "9223372036854775807" DAY);
+-----------------------------------------------------------+
| adddate("2018-12-09", interval "9223372036854775807" DAY) |
+-----------------------------------------------------------+
| NULL                                                      |
+-----------------------------------------------------------+
1 row in set, 1 warning (0.00 sec)

MySQL(root@127.0.0.1:test) > show warnings;
+---------+------+--------------------------------------------+
| Level   | Code | Message                                    |
+---------+------+--------------------------------------------+
| Warning | 1441 | Datetime function: datetime field overflow |
+---------+------+--------------------------------------------+
1 row in set (0.01 sec)
```

We should use `strconv.ParseFloat()` in this PR. `floatStrToIntStr` converts `1e1` to 10, which is not in MySQL. Another discovery is that TiDB does not properly handle the "**Datetime function: datetime field overflow**" warning:

```sql
TiDB(root@127.0.0.1:test) > select adddate("2018-12-09", interval "9223372036854775807" DAY);
+-----------------------------------------------------------+
| adddate("2018-12-09", interval "9223372036854775807" DAY) |
+-----------------------------------------------------------+
| 2018-12-09                                                |
+-----------------------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
You can do it in the next PR, this PR LGTM